server/lib/subscriptions.ts

```diff
@@ -27,13 +27,16 @@ export const updatePaymentMethodForSubscription = async (
   const newOrderData = { PaymentMethodId: newPaymentMethod.id, status: newStatus };
 
   // Subscription changes
-  let newSubscriptionData;
+  const newSubscriptionData = { isActive: true, deactivatedAt: null };
   const wasManagedExternally = getIsSubscriptionManagedExternally(prevPaymentMethod);
   const isManagedExternally = getIsSubscriptionManagedExternally(newPaymentMethod);
   if (wasManagedExternally !== isManagedExternally) {
-    newSubscriptionData = { isManagedExternally, paypalSubscriptionId: null };
+    newSubscriptionData['isManagedExternally'] = isManagedExternally;
+    newSubscriptionData['paypalSubscriptionId'] = null;
   }
 
+  // Need to cancel previous subscription
+  await order.Subscription.deactivate();
```

Comment:
This is the root fix, we must make sure the previous subscription is deactivated before activating the new one